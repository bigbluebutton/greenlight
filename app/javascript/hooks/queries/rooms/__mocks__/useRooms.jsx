export default function useRooms() {
  return {
    isLoading: false,
    data:
            [
              {
                name: 'Home Room',
                friendly_id: 'dummy_id1',
              },
              {
                name: 'Room 1',
                friendly_id: 'dummy_id2',
              },
              {
                name: 'Room 2',
                friendly_id: 'dummy_id3',
              },
            ],
  };
}
