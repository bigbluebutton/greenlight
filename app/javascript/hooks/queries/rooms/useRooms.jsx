import { useQuery } from 'react-query';
import axios from 'axios';

export default function useRooms() {
  return useQuery('getRooms', () => axios.get('/api/v1/rooms.json', {
    headers: {
      'Content-Type': 'application/json',
      Accept: 'application/json',
    },
    refetchInterval: 10000,
    refetchIntervalInBackground: true,
  }).then((resp) => resp.data.data));
}
