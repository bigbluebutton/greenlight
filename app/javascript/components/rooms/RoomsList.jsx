// BigBlueButton open source conferencing system - http://www.bigbluebutton.org/.
//
// Copyright (c) 2022 BigBlueButton Inc. and by respective authors (see below).
//
// This program is free software; you can redistribute it and/or modify it under the
// terms of the GNU Lesser General Public License as published by the Free Software
// Foundation; either version 3.0 of the License, or (at your option) any later
// version.
//
// Greenlight is distributed in the hope that it will be useful, but WITHOUT ANY
// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
// PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License along
// with Greenlight; if not, see <http://www.gnu.org/licenses/>.

import React, { useState } from 'react';
import {
  Row, Col, Button, Stack,
} from 'react-bootstrap';
import { useTranslation } from 'react-i18next';
import RoomCard from './RoomCard';
import useRooms from '../../hooks/queries/rooms/useRooms';
import useCreateRoom from '../../hooks/mutations/rooms/useCreateRoom';
import RoomCardPlaceHolder from './RoomCardPlaceHolder';
import Modal from '../shared_components/modals/Modal';
import CreateRoomForm from './room/forms/CreateRoomForm';
import { useAuth } from '../../contexts/auth/AuthProvider';
import SearchBar from '../shared_components/search/SearchBar';
import EmptyRoomsList from './EmptyRoomsList';
import NoSearchResults from '../shared_components/search/NoSearchResults';

export default function RoomsList() {
  const { t } = useTranslation();
  const [searchInput, setSearchInput] = useState('');
  const { isLoading, data: rooms } = useRooms(searchInput);
  const currentUser = useAuth();
  const mutationWrapper = (args) => useCreateRoom({ userId: currentUser.id, ...args });

  if (!isLoading && rooms?.length === 0 && !searchInput) {
    return <EmptyRoomsList />;
  }

  return (
    <>
      <Stack direction="horizontal" className="pt-5" gap={3}>
        <div>
          <SearchBar searchInput={searchInput} id="rooms-search" setSearchInput={setSearchInput} />
        </div>
        <Modal
          modalButton={(
            <Button
              variant="brand"
              className="ms-auto me-xxl-1"
            >{t('room.add_new_room')}
            </Button>
          )}
          title={t('room.create_new_room')}
          body={<CreateRoomForm mutation={mutationWrapper} userId={currentUser.id} />}
        />
      </Stack>
      <Row className="g-4 mt-4">
        {
          (isLoading && [...Array(8)].map((val, idx) => (
            <Col
              // eslint-disable-next-line react/no-array-index-key
              key={idx}
              className="col-md-auto mt-0 mb-4"
            ><RoomCardPlaceHolder />
            </Col>
          )))
          || (rooms?.length && rooms?.map((room) => (
            <Col key={room.friendly_id} className="col-md-auto col-xs-12 mt-0 mb-4">
              {(room.optimistic && <RoomCardPlaceHolder />) || <RoomCard room={room} />}
            </Col>
          )))
          || <NoSearchResults text={t('room.search_not_found')} searchInput={searchInput} />
        }
      </Row>
    </>
  );
}
