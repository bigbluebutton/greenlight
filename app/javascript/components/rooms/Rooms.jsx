// BigBlueButton open source conferencing system - http://www.bigbluebutton.org/.
//
// Copyright (c) 2022 BigBlueButton Inc. and by respective authors (see below).
//
// This program is free software; you can redistribute it and/or modify it under the
// terms of the GNU Lesser General Public License as published by the Free Software
// Foundation; either version 3.0 of the License, or (at your option) any later
// version.
//
// Greenlight is distributed in the hope that it will be useful, but WITHOUT ANY
// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
// PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License along
// with Greenlight; if not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import {
  Tab, Tabs,
} from 'react-bootstrap';
import { useTranslation } from 'react-i18next';
import RoomsList from './RoomsList';
import UserRecordings from '../recordings/UserRecordings';
import RecordingsCountTab from '../recordings/RecordingsCountTab';
import useRecordingsCount from '../../hooks/queries/recordings/useRecordingsCount';

export default function Rooms() {
  const { data: recordingsCount } = useRecordingsCount();
  const { t } = useTranslation();
  return (
    <Tabs className="wide-white pt-5" defaultActiveKey="rooms" unmountOnExit>
      <Tab className="background-whitesmoke" eventKey="rooms" title={t('room.rooms')}>
        <RoomsList />
      </Tab>
      <Tab className="background-whitesmoke" eventKey="recordings" title={<RecordingsCountTab count={recordingsCount} />}>
        <UserRecordings />
      </Tab>
    </Tabs>
  );
}
