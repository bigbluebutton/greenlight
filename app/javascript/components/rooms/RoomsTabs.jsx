import React from 'react';
import { Row, Tabs, Tab } from 'react-bootstrap';
import RecordingsTable from './RecordingsTable';
import RoomsList from './RoomsList';

export default function RoomsTabs() {
  return (
    <Row className="py-5">
      <Tabs defaultActiveKey="rooms">
        <Tab eventKey="rooms" title="Rooms">
          <RoomsList />
        </Tab>
        {/* TODO: May need to change this to it's own component depending on how RecordingsTable will work */}
        <Tab eventKey="recordings" title="Recordings">
          <RecordingsTable />
        </Tab>
      </Tabs>
    </Row>
  );
}
