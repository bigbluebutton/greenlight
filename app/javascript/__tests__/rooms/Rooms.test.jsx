import React from 'react';
import renderer from 'react-test-renderer';
import Rooms from '../../components/rooms/Rooms';
import RoomsList from '../../components/rooms/RoomsList';
import RoomsTabs from '../../components/rooms/RoomsTabs';
import useRooms from '../../hooks/queries/rooms/__mocks__/useRooms';

jest.mock('react-router-dom');
jest.mock('../../hooks/queries/rooms/useRooms');

const rooms = {
  data:
    [
      {
        name: 'Home Room',
        friendly_id: 'dummy_id1',
      },
      {
        name: 'Room 1',
        friendly_id: 'dummy_id2',
      },
      {
        name: 'Room 2',
        friendly_id: 'dummy_id3',
      },
    ],
};

describe('Rooms (useRooms query test)', () => {
  test('Rooms from query match expected rooms', () => {
    const json = useRooms();
    expect(json).toMatchObject(rooms);
  });
});

describe('Rooms (Snapshot Tests)', () => {
  it('Rooms renders rooms', () => {
    const roomsComponent = renderer.create(<Rooms />);
    const json = roomsComponent.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('RoomsTab renders room nav tabs', () => {
    const roomsTabsComponent = renderer.create(<RoomsTabs />);
    const json = roomsTabsComponent.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('RoomsList renders list of rooms', () => {
    const roomsListComponent = renderer.create(<RoomsList />);
    const json = roomsListComponent.toJSON();
    expect(json).toMatchSnapshot();
  });
});
