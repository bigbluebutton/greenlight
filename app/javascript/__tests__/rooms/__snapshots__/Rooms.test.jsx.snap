// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms (Snapshot Tests) Rooms renders rooms 1`] = `
<div
  className="container"
>
  <div
    className="py-5 row"
  >
    <ul
      className="nav nav-tabs"
      onKeyDown={[Function]}
      role="tablist"
    >
      <li
        className="nav-item"
        role="presentation"
      >
        <button
          aria-controls="react-aria2841199876-1-tabpane-rooms"
          aria-selected={true}
          className="nav-link active"
          data-rr-ui-event-key="rooms"
          disabled={false}
          id="react-aria2841199876-1-tab-rooms"
          onClick={[Function]}
          role="tab"
          type="button"
        >
          Rooms
        </button>
      </li>
      <li
        className="nav-item"
        role="presentation"
      >
        <button
          aria-controls="react-aria2841199876-1-tabpane-recordings"
          className="nav-link"
          data-rr-ui-event-key="recordings"
          disabled={false}
          id="react-aria2841199876-1-tab-recordings"
          onClick={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          Recordings
        </button>
      </li>
    </ul>
    <div
      className="tab-content"
    >
      <div
        aria-labelledby="react-aria2841199876-1-tab-rooms"
        className="fade tab-pane active show"
        id="react-aria2841199876-1-tabpane-rooms"
        role="tabpanel"
      >
        <div
          className="mt-4 mb-4 row"
        >
          <div
            className="col"
          >
            <form
              className="d-flex"
            >
              <input
                aria-label="Search"
                className="form-control"
                placeholder="Search"
                style={
                  Object {
                    "width": "19rem",
                  }
                }
                type="search"
              />
              <button
                className="ms-2 btn btn-outline-secondary"
                disabled={false}
                type="button"
              >
                Search
              </button>
            </form>
          </div>
          <div
            className="col"
          >
            <button
              className="float-end btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               + New Room 
            </button>
          </div>
        </div>
        <div
          className="g-4 row row-cols-md-4"
        >
          <div
            className="col"
          >
            <div
              className="rooms-card card border-light"
            >
              <div
                className="room-card-top pb-0 card-body"
                onClick={[Function]}
              >
                <svg
                  className="mb-4"
                  fill="currentColor"
                  height={55}
                  viewBox="0 0 16 16"
                  width={55}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
                  />
                  <path
                    d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
                  />
                </svg>
                <div
                  className="card-title h5"
                >
                   
                  Home Room
                   
                </div>
                <p
                  className="text-muted card-text"
                >
                   Last session... 
                </p>
                <hr />
              </div>
              <div
                className="pt-0 card-body"
              >
                <svg
                  fill="currentColor"
                  height={20}
                  viewBox="0 0 16 16"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
                  />
                  <path
                    d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
                  />
                </svg>
                <button
                  className="float-end btn btn-outline-secondary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                   Start
                </button>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="rooms-card card border-light"
            >
              <div
                className="room-card-top pb-0 card-body"
                onClick={[Function]}
              >
                <svg
                  className="mb-4"
                  fill="currentColor"
                  height={55}
                  viewBox="0 0 16 16"
                  width={55}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
                  />
                  <path
                    d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
                  />
                </svg>
                <div
                  className="card-title h5"
                >
                   
                  Room 1
                   
                </div>
                <p
                  className="text-muted card-text"
                >
                   Last session... 
                </p>
                <hr />
              </div>
              <div
                className="pt-0 card-body"
              >
                <svg
                  fill="currentColor"
                  height={20}
                  viewBox="0 0 16 16"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
                  />
                  <path
                    d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
                  />
                </svg>
                <button
                  className="float-end btn btn-outline-secondary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                   Start
                </button>
              </div>
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="rooms-card card border-light"
            >
              <div
                className="room-card-top pb-0 card-body"
                onClick={[Function]}
              >
                <svg
                  className="mb-4"
                  fill="currentColor"
                  height={55}
                  viewBox="0 0 16 16"
                  width={55}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
                  />
                  <path
                    d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
                  />
                </svg>
                <div
                  className="card-title h5"
                >
                   
                  Room 2
                   
                </div>
                <p
                  className="text-muted card-text"
                >
                   Last session... 
                </p>
                <hr />
              </div>
              <div
                className="pt-0 card-body"
              >
                <svg
                  fill="currentColor"
                  height={20}
                  viewBox="0 0 16 16"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
                  />
                  <path
                    d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
                  />
                </svg>
                <button
                  className="float-end btn btn-outline-secondary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                   Start
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="react-aria2841199876-1-tab-recordings"
        className="fade tab-pane"
        id="react-aria2841199876-1-tabpane-recordings"
        role="tabpanel"
      >
        <div
          className="bg-light row"
        >
          <p>
            Recordings Table
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Rooms (Snapshot Tests) RoomsList renders list of rooms 1`] = `
Array [
  <div
    className="mt-4 mb-4 row"
  >
    <div
      className="col"
    >
      <form
        className="d-flex"
      >
        <input
          aria-label="Search"
          className="form-control"
          placeholder="Search"
          style={
            Object {
              "width": "19rem",
            }
          }
          type="search"
        />
        <button
          className="ms-2 btn btn-outline-secondary"
          disabled={false}
          type="button"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col"
    >
      <button
        className="float-end btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
         + New Room 
      </button>
    </div>
  </div>,
  <div
    className="g-4 row row-cols-md-4"
  >
    <div
      className="col"
    >
      <div
        className="rooms-card card border-light"
      >
        <div
          className="room-card-top pb-0 card-body"
          onClick={[Function]}
        >
          <svg
            className="mb-4"
            fill="currentColor"
            height={55}
            viewBox="0 0 16 16"
            width={55}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
            />
            <path
              d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
            />
          </svg>
          <div
            className="card-title h5"
          >
             
            Home Room
             
          </div>
          <p
            className="text-muted card-text"
          >
             Last session... 
          </p>
          <hr />
        </div>
        <div
          className="pt-0 card-body"
        >
          <svg
            fill="currentColor"
            height={20}
            viewBox="0 0 16 16"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
            />
            <path
              d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
            />
          </svg>
          <button
            className="float-end btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
             Start
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="rooms-card card border-light"
      >
        <div
          className="room-card-top pb-0 card-body"
          onClick={[Function]}
        >
          <svg
            className="mb-4"
            fill="currentColor"
            height={55}
            viewBox="0 0 16 16"
            width={55}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
            />
            <path
              d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
            />
          </svg>
          <div
            className="card-title h5"
          >
             
            Room 1
             
          </div>
          <p
            className="text-muted card-text"
          >
             Last session... 
          </p>
          <hr />
        </div>
        <div
          className="pt-0 card-body"
        >
          <svg
            fill="currentColor"
            height={20}
            viewBox="0 0 16 16"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
            />
            <path
              d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
            />
          </svg>
          <button
            className="float-end btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
             Start
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="rooms-card card border-light"
      >
        <div
          className="room-card-top pb-0 card-body"
          onClick={[Function]}
        >
          <svg
            className="mb-4"
            fill="currentColor"
            height={55}
            viewBox="0 0 16 16"
            width={55}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
            />
            <path
              d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
            />
          </svg>
          <div
            className="card-title h5"
          >
             
            Room 2
             
          </div>
          <p
            className="text-muted card-text"
          >
             Last session... 
          </p>
          <hr />
        </div>
        <div
          className="pt-0 card-body"
        >
          <svg
            fill="currentColor"
            height={20}
            viewBox="0 0 16 16"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
            />
            <path
              d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
            />
          </svg>
          <button
            className="float-end btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
             Start
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Rooms (Snapshot Tests) RoomsTab renders room nav tabs 1`] = `
<div
  className="py-5 row"
>
  <ul
    className="nav nav-tabs"
    onKeyDown={[Function]}
    role="tablist"
  >
    <li
      className="nav-item"
      role="presentation"
    >
      <button
        aria-controls="react-aria2841199876-2-tabpane-rooms"
        aria-selected={true}
        className="nav-link active"
        data-rr-ui-event-key="rooms"
        disabled={false}
        id="react-aria2841199876-2-tab-rooms"
        onClick={[Function]}
        role="tab"
        type="button"
      >
        Rooms
      </button>
    </li>
    <li
      className="nav-item"
      role="presentation"
    >
      <button
        aria-controls="react-aria2841199876-2-tabpane-recordings"
        className="nav-link"
        data-rr-ui-event-key="recordings"
        disabled={false}
        id="react-aria2841199876-2-tab-recordings"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Recordings
      </button>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    <div
      aria-labelledby="react-aria2841199876-2-tab-rooms"
      className="fade tab-pane active show"
      id="react-aria2841199876-2-tabpane-rooms"
      role="tabpanel"
    >
      <div
        className="mt-4 mb-4 row"
      >
        <div
          className="col"
        >
          <form
            className="d-flex"
          >
            <input
              aria-label="Search"
              className="form-control"
              placeholder="Search"
              style={
                Object {
                  "width": "19rem",
                }
              }
              type="search"
            />
            <button
              className="ms-2 btn btn-outline-secondary"
              disabled={false}
              type="button"
            >
              Search
            </button>
          </form>
        </div>
        <div
          className="col"
        >
          <button
            className="float-end btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
             + New Room 
          </button>
        </div>
      </div>
      <div
        className="g-4 row row-cols-md-4"
      >
        <div
          className="col"
        >
          <div
            className="rooms-card card border-light"
          >
            <div
              className="room-card-top pb-0 card-body"
              onClick={[Function]}
            >
              <svg
                className="mb-4"
                fill="currentColor"
                height={55}
                viewBox="0 0 16 16"
                width={55}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
                />
                <path
                  d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
                />
              </svg>
              <div
                className="card-title h5"
              >
                 
                Home Room
                 
              </div>
              <p
                className="text-muted card-text"
              >
                 Last session... 
              </p>
              <hr />
            </div>
            <div
              className="pt-0 card-body"
            >
              <svg
                fill="currentColor"
                height={20}
                viewBox="0 0 16 16"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
                />
                <path
                  d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
                />
              </svg>
              <button
                className="float-end btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                 Start
              </button>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="rooms-card card border-light"
          >
            <div
              className="room-card-top pb-0 card-body"
              onClick={[Function]}
            >
              <svg
                className="mb-4"
                fill="currentColor"
                height={55}
                viewBox="0 0 16 16"
                width={55}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
                />
                <path
                  d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
                />
              </svg>
              <div
                className="card-title h5"
              >
                 
                Room 1
                 
              </div>
              <p
                className="text-muted card-text"
              >
                 Last session... 
              </p>
              <hr />
            </div>
            <div
              className="pt-0 card-body"
            >
              <svg
                fill="currentColor"
                height={20}
                viewBox="0 0 16 16"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
                />
                <path
                  d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
                />
              </svg>
              <button
                className="float-end btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                 Start
              </button>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="rooms-card card border-light"
          >
            <div
              className="room-card-top pb-0 card-body"
              onClick={[Function]}
            >
              <svg
                className="mb-4"
                fill="currentColor"
                height={55}
                viewBox="0 0 16 16"
                width={55}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
                />
                <path
                  d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm12 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1v-1c0-1-1-4-6-4s-6 3-6 4v1a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12z"
                />
              </svg>
              <div
                className="card-title h5"
              >
                 
                Room 2
                 
              </div>
              <p
                className="text-muted card-text"
              >
                 Last session... 
              </p>
              <hr />
            </div>
            <div
              className="pt-0 card-body"
            >
              <svg
                fill="currentColor"
                height={20}
                viewBox="0 0 16 16"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
                />
                <path
                  d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
                />
              </svg>
              <button
                className="float-end btn btn-outline-secondary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                 Start
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="react-aria2841199876-2-tab-recordings"
      className="fade tab-pane"
      id="react-aria2841199876-2-tabpane-recordings"
      role="tabpanel"
    >
      <div
        className="bg-light row"
      >
        <p>
          Recordings Table
        </p>
      </div>
    </div>
  </div>
</div>
`;
